test/topology_experimental_raft/test_mv_tablets.py

```diff
@@ -7,14 +7,74 @@
 # Tests for interaction of materialized views with *tablets*
 
 from test.pylib.manager_client import ManagerClient
+from test.pylib.util import wait_for_cql_and_get_hosts, read_barrier
+from test.pylib.internal_types import ServerInfo
+from test.topology.conftest import skip_mode
 
 import pytest
 import asyncio
 import logging
+import random
+import time
 
 
 logger = logging.getLogger(__name__)
 
+# This convenience function takes the name of a table or a view, and a token,
+# and returns the list of host_id,shard pairs holding tablets for this token
+# and view.
+# You also need to specify a specific server to use for the requests, to
+# ensure that if you send tablet-migration commands to one server, you also
+# read the replicas information from the same server (it takes time for this
+# information to propagate to all servers).
+async def get_tablet_replicas(manager: ManagerClient, server: ServerInfo, keyspace_name: str, table_or_view_name: str, token: int):
```

Comment:
This is exactly what I'm trying to avoid - random test developers coming up with what they think **might** be some useful general-purpose functions and putting those functions in some "global" place, and then when the next test developer comes along, they discover they need slight (or large) variations on these functions. You can see the result in dtest - a large number of functions that nobody understands and have dozens of options each.

Please note that this function also appears in test_tablets.py (written by @tgrabiec), but in a different form - it didn't support materialized views. So 1. Tomek also didn't put this in manager, 2. It demonstrates my point that a different test couldn't use the same function exactly, and had to modify it anyway.

Eventually, it will be nice to take some common useful functions and move them into a central place, but I don't want to do it too early. Also, this has **absolutely no business** being in the "manager". The "manager" is about managing a cluster of nodes - we don't need to stuff collections of utility functions into the manager, unless we want the result to be a huge monolithic mess.

Reply:
> Please note that this function also appears in test_tablets.py (written by @tgrabiec), but in a different form - it didn't support materialized views. So 1. Tomek also didn't put this in manager, 2. It demonstrates my point that a different test couldn't use the same function exactly, and had to modify it anyway.

I think this function is actually a counter example. get_tablet_replicas() is something which many tablet test cases will use, we should just make it work for all kinds of tables. The first usage which didn't take views into account won't mind.

Agree that the manager is not the best place.

Reply:
In my most recent version (I'll push it in a few minutes), I strayed even further from your original implementation - I no longer need the extra get_table_id() and instead do a full-table scan to look for the right table without knowing its id. The result is much simpler, but I don't know if you'd like to use it or not. It's also **six lines of code**. It's not a library, it's not a 300-line implementation of some traditional data structure you learn in CS courses - it's an ad-hoc 6-line function. I don't want to decide right now if it should be shared with other tests or not. I don't have to decide it right now. If later these 6-line function is determined to be very useful, we can move it to a common place.